contracts/proxies/GnosisSafeProxyFactory.sol

```diff
@@ -63,7 +58,27 @@ contract GnosisSafeProxyFactory {
         bytes memory initializer,
         uint256 saltNonce
     ) public returns (GnosisSafeProxy proxy) {
-        proxy = deployProxyWithNonce(_singleton, initializer, saltNonce);
+        proxy = deployProxyWithNonce(_singleton, initializer, saltNonce, false);
+        if (initializer.length > 0)
+            // solhint-disable-next-line no-inline-assembly
+            assembly {
+                if eq(call(gas(), proxy, 0, add(initializer, 0x20), mload(initializer), 0, 0), 0) {
+                    revert(0, 0)
+                }
+            }
+        emit ProxyCreation(proxy, _singleton);
+    }
+
+    /// @dev Same as `createProxyWithNonce` but includes the chain id in the salt to calculate the address of the new proxy contract.
```

Comment:
It probably would be nice to add proper natspec (with placeholders for the params).